pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/OwnershipCache.java

```diff
@@ -170,7 +170,17 @@ public CompletableFuture<Optional<NamespaceEphemeralData>> getOwnerAsync(Namespa
 
         // If we're not the owner, we need to check if anybody else is
         String path = ServiceUnitUtils.path(suName);
-        return lockManager.readLock(path);
+        return lockManager.readLock(path).thenCompose(owner -> {
+            //If the current broker is the owner, we must try to acquire ownership to avoid cache loss.
+            if (owner.isPresent() && owner.get().equals(selfOwnerInfo)) {
+                try {
+                    return tryAcquiringOwnership(suName).thenApply(Optional::ofNullable);
```

Comment:
As this shouldn't happen in normal cases. Please help add a warning log here.

Reply:
Ok, good catch!